### ISSUE TITLE
FreshAnimations compat

### ISSUE THREAD

**Issue by hypercross** [Christofmeg/JustEnoughBreeding#18]
![image](https://github.com/user-attachments/assets/d37eda71-8848-4ef9-86e9-74705089caf6)

Mob renders oddly for a moment every 2 seconds or so when FreshAnimations is enabled...

**Comment by Christofmeg** [comment]
* Which modloader are you using?
* Which version of the game are you running?
* Which version of Fresh Animations do you have installed?
* Which version of Just Enough Breeding are you using?

**Comment by hypercross** [comment]
Minecraft `1.20.1`, Fabric `0.15.11`, FreshAnimations `1.9.2`, Just Enough Breeding `1.2.4`

Here's the mod list:
```
[20:19:33] [main/INFO]: Loading Minecraft 1.20.1 with Fabric Loader 0.15.11
[20:19:33] [ForkJoinPool-1-worker-30/WARN]: Mod irisflw uses the version 1.1.3a which isn't compatible with Loader's extended semantic version format (Could not parse version number component '3a'!), SemVer is recommended for reliably evaluating dependencies and prioritizing newer version
[20:19:33] [ForkJoinPool-1-worker-10/WARN]: Mod tectonic uses the version 2.3.5b which isn't compatible with Loader's extended semantic version format (Could not parse version number component '5b'!), SemVer is recommended for reliably evaluating dependencies and prioritizing newer version
[20:19:34] [main/INFO]: Loading 184 mods:
	- antique-atlas 2.9.13+1.20
	   \-- surveyor 0.6.21+1.20
	        \-- folk_sisby_kaleido-config 0.3.1+1.3.1
	- appleskin 2.5.1+mc1.20
	- bettercombat 1.8.6+1.20.1
	- betterdeserttemples 1.20-Fabric-3.0.3
	   \-- org_reflections_reflections 0.10.2
	- betterdungeons 1.20-Fabric-4.0.4
	- betterfortresses 1.20-Fabric-2.0.6
	- betterjungletemples 1.20-Fabric-2.0.5
	- bettermineshafts 1.20-Fabric-4.0.4
	- betteroceanmonuments 1.20-Fabric-3.0.4
	- betterstrongholds 1.20-Fabric-4.0.3
	- betterwitchhuts 1.20-Fabric-3.0.3
	- c2me 0.2.0+alpha.11.13
	   |-- c2me-base 0.2.0+alpha.11.13
	   |-- c2me-client-uncapvd 0.2.0+alpha.11.13
	   |-- c2me-fixes-chunkio-threading-issues 0.2.0+alpha.11.13
	   |-- c2me-fixes-general-threading-issues 0.2.0+alpha.11.13
	   |-- c2me-fixes-worldgen-threading-issues 0.2.0+alpha.11.13
	   |-- c2me-fixes-worldgen-vanilla-bugs 0.2.0+alpha.11.13
	   |-- c2me-notickvd 0.2.0+alpha.11.13
	   |-- c2me-opts-allocs 0.2.0+alpha.11.13
	   |-- c2me-opts-chunk-access 0.2.0+alpha.11.13
	   |-- c2me-opts-chunkio 0.2.0+alpha.11.13
	   |-- c2me-opts-math 0.2.0+alpha.11.13
	   |-- c2me-opts-scheduling 0.2.0+alpha.11.13
	   |-- c2me-opts-worldgen-general 0.2.0+alpha.11.13
	   |-- c2me-opts-worldgen-vanilla 0.2.0+alpha.11.13
	   |-- c2me-rewrites-chunk-serializer 0.2.0+alpha.11.13
	   |-- c2me-rewrites-chunkio 0.2.0+alpha.11.13
	   |-- c2me-server-utils 0.2.0+alpha.11.13
	   |-- c2me-threading-chunkio 0.2.0+alpha.11.13
	   |-- c2me-threading-lighting 0.2.0+alpha.11.13
	   |-- c2me-threading-worldgen 0.2.0+alpha.11.13
	   |-- com_ibm_async_asyncutil 0.1.0
	   |-- mixinextras 0.3.6
	   |-- net_objecthunter_exp4j 0.4.8
	   \-- org_threadly_threadly 7.0
	- cloth-config 11.1.118
	   \-- cloth-basic-math 0.6.1
	- continuity 3.0.0-beta.5+1.20.1
	- copycats 2.1.2+mc.1.20.1-fabric
	- create 0.5.1-f-build.1417+mc1.20.1
	   |-- com_electronwill_night-config_core 3.6.6
	   |-- com_electronwill_night-config_toml 3.6.6
	   |-- com_google_code_findbugs_jsr305 3.0.2
	   |-- flywheel 0.6.10-2
	   |-- milk 1.2.60
	   |    \-- dripstone_fluid_lib 3.0.2
	   |-- porting_lib_brewing 2.3.2+1.20.1
	   |-- porting_lib_models 2.3.2+1.20.1
	   |    \-- porting_lib_model_loader 2.3.2+1.20.1
	   |-- porting_lib_obj_loader 2.3.2+1.20.1
	   |    \-- porting_lib_model_loader 2.3.2+1.20.1
	   |-- porting_lib_tags 3.0
	   |-- reach-entity-attributes 2.4.0
	   \-- registrate-fabric 1.3.62-MC1.20.1
	        |-- porting_lib_data 2.1.1090+1.20
	        |    \-- porting_lib_gametest 2.1.1090+1.20
	        \-- porting_lib_model_generators 2.1.1090+1.20
	             |-- porting_lib_gametest 2.1.1090+1.20
	             \-- porting_lib_model_materials 2.1.1090+1.20
	- customskinloader 14.20
	- distanthorizons 2.1.2-a
	- emi 1.1.12+1.20.1+fabric
	- enhancedblockentities 0.9+1.20
	   |-- advanced_runtime_resource_pack 0.6.7
	   \-- spruceui 5.0.0+1.20
	- entity_model_features 2.2
	- entity_texture_features 6.2
	   \-- org_apache_httpcomponents_httpmime 4.5.10
	- entityculling 1.6.7
	- fabric-api 0.92.2+1.20.1
	   |-- fabric-api-base 0.4.31+1802ada577
	   |-- fabric-api-lookup-api-v1 1.6.36+1802ada577
	   |-- fabric-biome-api-v1 13.0.13+1802ada577
	   |-- fabric-block-api-v1 1.0.11+1802ada577
	   |-- fabric-block-view-api-v2 1.0.1+1802ada577
	   |-- fabric-blockrenderlayer-v1 1.1.41+1802ada577
	   |-- fabric-client-tags-api-v1 1.1.2+1802ada577
	   |-- fabric-command-api-v1 1.2.34+f71b366f77
	   |-- fabric-command-api-v2 2.2.13+1802ada577
	   |-- fabric-commands-v0 0.2.51+df3654b377
	   |-- fabric-containers-v0 0.1.64+df3654b377
	   |-- fabric-content-registries-v0 4.0.11+1802ada577
	   |-- fabric-convention-tags-v1 1.5.5+1802ada577
	   |-- fabric-crash-report-info-v1 0.2.19+1802ada577
	   |-- fabric-data-attachment-api-v1 1.0.0+de0fd6d177
	   |-- fabric-data-generation-api-v1 12.3.4+1802ada577
	   |-- fabric-dimensions-v1 2.1.54+1802ada577
	   |-- fabric-entity-events-v1 1.6.0+1c78457f77
	   |-- fabric-events-interaction-v0 0.6.2+1802ada577
	   |-- fabric-events-lifecycle-v0 0.2.63+df3654b377
	   |-- fabric-game-rule-api-v1 1.0.40+1802ada577
	   |-- fabric-item-api-v1 2.1.28+1802ada577
	   |-- fabric-item-group-api-v1 4.0.12+1802ada577
	   |-- fabric-key-binding-api-v1 1.0.37+1802ada577
	   |-- fabric-keybindings-v0 0.2.35+df3654b377
	   |-- fabric-lifecycle-events-v1 2.2.22+1802ada577
	   |-- fabric-loot-api-v2 1.2.1+1802ada577
	   |-- fabric-loot-tables-v1 1.1.45+9e7660c677
	   |-- fabric-message-api-v1 5.1.9+1802ada577
	   |-- fabric-mining-level-api-v1 2.1.50+1802ada577
	   |-- fabric-model-loading-api-v1 1.0.3+1802ada577
	   |-- fabric-models-v0 0.4.2+9386d8a777
	   |-- fabric-networking-api-v1 1.3.11+1802ada577
	   |-- fabric-networking-v0 0.3.51+df3654b377
	   |-- fabric-object-builder-api-v1 11.1.3+1802ada577
	   |-- fabric-particles-v1 1.1.2+1802ada577
	   |-- fabric-recipe-api-v1 1.0.21+1802ada577
	   |-- fabric-registry-sync-v0 2.3.3+1802ada577
	   |-- fabric-renderer-api-v1 3.2.1+1802ada577
	   |-- fabric-renderer-indigo 1.5.2+85287f9f77
	   |-- fabric-renderer-registries-v1 3.2.46+df3654b377
	   |-- fabric-rendering-data-attachment-v1 0.3.37+92a0d36777
	   |-- fabric-rendering-fluids-v1 3.0.28+1802ada577
	   |-- fabric-rendering-v0 1.1.49+df3654b377
	   |-- fabric-rendering-v1 3.0.8+1802ada577
	   |-- fabric-resource-conditions-api-v1 2.3.8+1802ada577
	   |-- fabric-resource-loader-v0 0.11.10+1802ada577
	   |-- fabric-screen-api-v1 2.0.8+1802ada577
	   |-- fabric-screen-handler-api-v1 1.3.30+1802ada577
	   |-- fabric-sound-api-v1 1.0.13+1802ada577
	   |-- fabric-transfer-api-v1 3.3.5+8dd72ea377
	   \-- fabric-transitive-access-wideners-v1 4.3.1+1802ada577
	- fabricloader 0.15.11
	- farmersdelight 1.20.1-2.1.3+refabricated
	   |-- mm 2.3
	   |-- porting_lib_accessors 2.3.4+1.20.1
	   |    \-- porting_lib_core 2.3.4+1.20.1
	   |-- porting_lib_base 2.3.4+1.20.1
	   |    |-- com_electronwill_night-config_core 3.6.6
	   |    |-- com_electronwill_night-config_toml 3.6.6
	   |    |-- porting_lib_attributes 2.3.4+1.20.1
	   |    |-- porting_lib_common 2.3.4+1.20.1
	   |    |-- porting_lib_core 2.3.4+1.20.1
	   |    |-- porting_lib_entity 2.3.4+1.20.1
	   |    |-- porting_lib_fluids 2.3.4+1.20.1
	   |    |-- porting_lib_mixin_extensions 2.3.4+1.20.1
	   |    |-- porting_lib_transfer 2.3.4+1.20.1
	   |    |-- porting_lib_utility 2.3.4+1.20.1
	   |    \-- reach-entity-attributes 2.4.0
	   |-- porting_lib_client_events 2.3.4+1.20.1
	   |    \-- porting_lib_core 2.3.4+1.20.1
	   |-- porting_lib_config 2.3.4+1.20.1
	   |    |-- com_electronwill_night-config_core 3.6.6
	   |    |-- com_electronwill_night-config_toml 3.6.6
	   |    \-- porting_lib_core 2.3.4+1.20.1
	   |-- porting_lib_extensions 2.3.4+1.20.1
	   |    |-- porting_lib_attributes 2.3.4+1.20.1
	   |    |-- porting_lib_common 2.3.4+1.20.1
	   |    |-- porting_lib_core 2.3.4+1.20.1
	   |    \-- reach-entity-attributes 2.4.0
	   |-- porting_lib_lazy_registration 2.3.4+1.20.1
	   |    \-- porting_lib_core 2.3.4+1.20.1
	   |-- porting_lib_loot 2.3.4+1.20.1
	   |    \-- porting_lib_core 2.3.4+1.20.1
	   |-- porting_lib_networking 2.3.4+1.20.1
	   |    \-- porting_lib_core 2.3.4+1.20.1
	   |-- porting_lib_recipe_book_categories 2.3.4+1.20.1
	   |    \-- porting_lib_core 2.3.4+1.20.1
	   |-- porting_lib_registries 2.3.4+1.20.1
	   |    \-- porting_lib_core 2.3.4+1.20.1
	   \-- porting_lib_tool_actions 2.3.4+1.20.1
	        \-- porting_lib_core 2.3.4+1.20.1
	- ferritecore 6.0.1
	- forgeconfigapiport 8.0.0
	- forgeconfigscreens 8.0.2
	- fxntstorage 0.18-1.20.1-FABRIC+1.20.1
	- immediatelyfast 1.2.20+1.20.4
	   \-- net_lenni0451_reflect 1.3.4
	- incendium 5.3.5
	- indium 1.0.34+mc1.20.1
	- iris 1.7.2+mc1.20.1
	   |-- fabric-api-base 0.4.31+1802ada577
	   |-- fabric-key-binding-api-v1 1.0.37+1802ada577
	   |-- io_github_douira_glsl-transformer 2.0.1
	   |-- org_anarres_jcpp 1.4.14
	   \-- org_antlr_antlr4-runtime 4.13.1
	- irisflw 1.1.3a
	   \-- flywheel 0.6.10-2
	- jade 11.10.0+fabric
	- java 17
	- justenoughbreeding 1.2.4
	- minecraft 1.20.1
	- modernfix 5.19.1+mc1.20.1
	- modmenu 7.2.2
	- moonlight 1.20-2.12.11
	- noisium 2.3.0+mc1.20-1.20.1
	- nullscape 1.2.7
	- player-animator 1.0.2-rc1+1.20
	- puzzleslib 8.1.21
	   \-- puzzlesaccessapi 8.0.7
	- railways 1.6.4+fabric-mc1.20.1
	   \-- mm 2.3
	- seasons 2.3+1.20
	- seasonsdelightcompat 1.0-1.3.9
	- seasonsterralithcompat 1.0-2.3.7
	- sodium 0.5.11+mc1.20.1
	   |-- fabric-api-base 0.4.31+1802ada577
	   |-- fabric-block-view-api-v2 1.0.1+1802ada577
	   |-- fabric-rendering-data-attachment-v1 0.3.37+92a0d36777
	   |-- fabric-rendering-fluids-v1 3.0.28+1802ada577
	   \-- fabric-resource-loader-v0 0.11.10+1802ada577
	- spelunkery 1.20.1-0.3.10
	- tectonic 2.3.5b
	- terralith 2.5.4
	- treechop 0.19.0
	- yungsapi 1.20-Fabric-4.0.5
	   \-- org_javassist_javassist 3.29.2-GA
	- yungsbridges 1.20-Fabric-4.0.3
	- yungsextras 1.20-Fabric-4.0.3
```

**Comment by Christofmeg** [comment]
This is the same as https://github.com/Christofmeg/JustEnoughBreeding/issues/14
> I did some testing and the issue only occurs when a resourcepack changes vanilla mob models with .jem files. EMF allows you to load in .jem files from optifine.
> 
> Tested with Optifine instead of [EMF] Entity Model Features and [ETF] Entity Texture Features and the issue went away. That indicates that you should file a bug report on the [EMF](https://github.com/Traben-0/Entity_Model_Features/issues) github. And hopefully you can get more help there.

The model glitching occurs only the first time EMF renders the entity.

To address this, Just Enough Breeding version 1.3.1 will include a change to prevent the category from generating a new entity every 5 seconds when EMF or Optifine is installed. With this update, if EMF is used with a resource pack that modifies mob models, the glitch will only appear once, during the initial rendering of the mob in that world session.

You should still report this issue to [EMF](https://github.com/Traben-0/Entity_Model_Features/issues)  if it still bugs you after version 1.3.1 is out.
